lightning-background-processor/src/lib.rs

```diff
@@ -491,9 +491,10 @@ mod tests {
 
 		macro_rules! check_persisted_data {
 			($node: expr, $filepath: expr, $expected_bytes: expr) => {
-				match $node.write(&mut $expected_bytes) {
-					Ok(()) => {
-						loop {
+				loop {
+					$expected_bytes.clear();
+					match $node.write(&mut $expected_bytes) {
+						Ok(()) => {
 							match std::fs::read($filepath) {
 								Ok(bytes) => {
 									if bytes == $expected_bytes {
```

Comment:
When I was debugging the CI hang, I got as far as figuring out that this is the line that it loops on repeatedly (unless it's repeatedly erroring on `fs::read`). So, I think this patch would fix CI (though IDK what problem that might indicate in the `FilesystemPersister`)

Reply:
At least for the purposes of this PR, the issue is that the `ChannelManager` can now change out from under us - the `ChannelManager` changes by a byte or two after one or two timer ticks, and thus the original version written at the top of the loop isn't necessarily the version that gets written to disk. It's not actually an issue, though.